utils/pythonutils/poetryutils.go

```diff
@@ -116,9 +125,105 @@ func extractPackagesFromPoetryLock(lockFilePath string) (dependencies map[string
 	dependenciesVersions = make(map[string]string)
 	dependencies = make(map[string][]string)
 	for _, dependency := range poetryLockFile.Package {
-		dependenciesVersions[dependency.Name] = dependency.Version
+		dependenciesVersions[strings.ToLower(dependency.Name)] = dependency.Version
 		dependencyName := dependency.Name + ":" + dependency.Version
 		dependencies[dependencyName] = maps.Keys(dependency.Dependencies)
 	}
 	return
 }
+
+func extractPoetryDependenciesFiles(srcPath string, cmdArgs []string, log utils.Log) (dependenciesFiles map[string]entities.Dependency, err error) {
```

Comment:
* Let's add a description for this function, specifically describing the returned values and the structure of `dependenciesFiles`.
* Let's add a unit test for this function.
* I also suggest adding some more focused tests for the `getSitePackagesPath` function.

Reply:
Tests will be added in following PR.